homeassistant/components/sonos/speaker.py

```diff
@@ -278,27 +282,71 @@ async def _subscribe(
         subscription.auto_renew_fail = self.async_renew_failed
         self._subscriptions.append(subscription)
 
-    @callback
-    def async_dispatch_properties(self, event: SonosEvent | None = None) -> None:
-        """Update properties from event."""
-        self.hass.async_create_task(self.async_update_device_properties(event))
-
-    @callback
-    def async_dispatch_alarms(self, event: SonosEvent | None = None) -> None:
-        """Update alarms from event."""
-        self.hass.async_create_task(self.async_update_alarms(event))
+    @property
+    def event_dispatchers(self) -> dict[str, Callable]:
+        """Return a dispatching dict for event routing."""
+        return {
```

Comment:
Maybe create this in `__init__` as `self._event_dispatchers` so the `dict` isn't recreated every time the property is called?